src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,142 @@
+ * For their functionality see the corresponding comments.
+ * @function checkInput then filters out all wrong @param word
+ * from @param text and puts them all into @param allWrongWords,
+ * which is printed in the console log by calling @function logInvalidWords.
+ * This is for debugging purposes and should later be changed to
+ * be displayed to the user. For further postprocessing all wrong
+ * words are collected in a @returns @param invalidWords of
+ * @type string array.
+ */
+
+export function checkInput(text: string) {
+		const preprocessedText: string[] = preprocessText(text);
+		const invalidWords = preprocessedText.filter(word => !allowedWords.includes(word));
+		// TODO: Make logInvalidWords dependent on this posOfInvalidWords.
+		const mapOfInvalidWords = createMapOfInvalidWords(text, invalidWords);
+		logInvalidWords(mapOfInvalidWords);
+		return invalidWords;
+}
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns a map of invalid words (consisted in @param invalidWords) and their corresponding
+ * positions (starting position and end position) inside of a string.
+ */
+
+export function createMapOfInvalidWords(text: string, invalidWords: string[]) {
+	const positions = new Map();
+	for (const word of invalidWords) {
+		positions.set(word, [text.indexOf(word), text.indexOf(word) + word.length - 1]);
+	}
+	return positions;
+}
+
+/**
+ * (input:) @param text
+ * This method deletes all UNICODE-characters \u2000-\u206F\u2E00-\u2E7F
+ * as well as '!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/. and
+ * replaces them with whitespaces. A word is afterwards (in @function splitIntoWords)
+ * considered as a substring with has at least a whitespace on the right
+ * or on the left.
+ * The exact symbols that can be deleted may have to be overthought,
+ * since brackets, commas and fullstops seem to be necessary to make the user input Prolog-readable.
+ */
+
+export function eliminatePunctuation(text: string) {
+		return text.replace(punctuation, " ").replace(space, " ");
+}
+
+/**
+ * (input:) @param text
+ * This method first deletes and leading or trailing whitespaces off of the
+ * input string. It then calls @function findSpacePositionsInStringArray, to
+ * create an array containing the positions where a new word begins.
+ * Currently this adds a whitespace to every word after the first one.
+ * After doing some corrections to the text, the method @returns @param result,
+ * which is an array containing as entries the single words filtered out
+ * of the text.
+ */
+
+export function splitIntoWords(text: string) {
+		text = text.trim();
+		const spacePositions = findSpacePositionsInStringArray(text);
+		let tempPos = 0;
+		const result: string[] = [];
+				// this loop is problematic. Adds a whitespace to the beginning of each word after the second one.
+		spacePositions.forEach(element => {
+				result.push(text.substring(tempPos, element));
+				tempPos = element;
+		});
```

Comment:
Please do this all with a regex instead that matches words surrounded by word barriers. After that, this function should be a simple loop. And you can even get the index of your findings that way.

Reply:
I now played around with some regular expressions and got /\b(\w*)\w\b/g, but how does that help me to get the positions  of the word barriers numerically?